### PR TITLE
Move C and Java implementation of murmurhash inside the gem

### PR DESCRIPTION
**Pull request by maksimf**
This PR is WIP

This solves #191

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -1,4 +1,24 @@
-require "bundler/gem_tasks"
-task :default => :spec
+require 'bundler/gem_tasks'
```
Comment (patricioe) [inline Rakefile]: @maksimf does this have any implications on our customers build tools?

File: Rakefile [hunk 2]
```diff
@@ -1,4 +1,24 @@
...
+RSpec::Core::RakeTask.new(:spec)
 
-Dir.glob('tasks/**/*.rake').each(&method(:import))
+task :spec => :compile
+case RUBY_PLATFORM
+when 'java'
+  require 'rake/javaextensiontask'
+  Rake::JavaExtensionTask.new 'murmurhash' do |ext|
+    ext.lib_dir = 'lib/murmurhash'
+    ext.target_version = '1.7'
```
Comment (patricioe) [inline Rakefile]: is this = to 1.7 or can it be greater than 1.7?

what's the minimum version of java we can support?

File: lib/splitclient-rb/engine/evaluator/splitter.rb [hunk 3]
```diff
@@ -50,7 +48,12 @@ def count_hash(key, seed, legacy)
       end
 
       def murmur_hash(key, seed)
-        Digest::MurmurHash3_x86_32.rawdigest(key, [seed].pack('L'))
+        case RUBY_PLATFORM
```
Comment (patricioe) [inline lib/splitclient-rb/engine/evaluator/splitter.rb]: what's the runtime penalty?

Can we avoid making this case statement on every evaluation and bind it one time during start up?

File: Rakefile (hunk 1 above)
Comment (maksimf) [reply Rakefile]: What do you mean? I just changed default rake task not to be `spec`

File: Rakefile (hunk 2 above)
Comment (maksimf) [reply Rakefile]: I had compilation issues with default version (1.6)

File: lib/splitclient-rb/engine/evaluator/splitter.rb (hunk 3 above)
Comment (maksimf) [reply lib/splitclient-rb/engine/evaluator/splitter.rb]: Yes, thank you, I'll change that

Comment (maksimf) [reply lib/splitclient-rb/engine/evaluator/splitter.rb]: @patricioe fixed

File: Rakefile (hunk 2 above)
Comment (patricioe) [reply Rakefile]: is 1.7 the minimum version? or does it have to match the exact version?
I want to make sure it will work with 1.8 as well.

Comment (maksimf) [reply Rakefile]: It is the minimum version. By default it was compiling for 1.5, but there were compilation issues and error for 1.5 and warnings for 1.6:

1.5:

```
javac  -target 1.5 -source 1.5 -Xlint:unchecked  -cp "/Users/max/.rbenv/versions/jruby-9.1.15.0/lib/jruby.jar:" -d tmp/java/murmurhash ext/murmurhash/MurmurHash3.java
warning: [options] bootstrap class path not set in conjunction with -source 1.5
error: Source option 1.5 is no longer supported. Use 1.6 or later.
error: Target option 1.5 is no longer supported. Use 1.6 or later.
```

1.6:

```
javac  -target 1.6 -source 1.6 -Xlint:unchecked  -cp "/Users/max/.rbenv/versions/jruby-9.1.15.0/lib/jruby.jar:" -d tmp/java/murmurhash ext/murmurhash/MurmurHash3.java
warning: [options] bootstrap class path not set in conjunction with -source 1.6
warning: [options] source value 1.6 is obsolete and will be removed in a future release
warning: [options] target value 1.6 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
```

So, I decided to use minimum of 1.7.

**Pull request merged**